lib/Doctrine/Migrations/Finder/RecursiveRegexFinder.php

```diff
@@ -14,8 +14,20 @@
 /**
  * The RecursiveRegexFinder class recursively searches the given directory for migrations.
  */
-final class RecursiveRegexFinder extends Finder implements MigrationDeepFinder
+final class RecursiveRegexFinder extends Finder
 {
+    /** @var string */
+    private $pattern;
+
+    public function __construct(?string $pattern = null)
+    {
+        $this->pattern = $pattern ?: sprintf(
+            '#^.+\\%sVersion[^\\%s]{1,255}\\.php$#i',
```

Comment:
👍🏻 for the warning

Reply:
What about allowing any class name? (since the sorting is not a problem anymore?)
Consider that this is not a limitation I have introduced, it was already present in 2.x (see https://github.com/doctrine/migrations/blob/1f9e877ef5a9b5403fc6581449f3fe55c8518500/lib/Doctrine/Migrations/Finder/RecursiveRegexFinder.php#L47)

Reply:
Yeah, that's what I thought. If there's no reason to keep the limit, let's drop it.

Reply:
done